packages/typespec-client-generator-core/src/internal-utils.ts

```diff
@@ -257,9 +260,69 @@ export function getSdkTypeBaseHelper<TKind>(
     __raw: type,
     deprecation: getDeprecationDetails(context.program, type)?.message,
     kind,
+    decorators: getTypeDecorators(context, type),
   };
 }
 
+export function getTypeDecorators(context: TCGCContext, type: Type): Record<string, Array<any>> {
+  const retval: Record<string, Array<any>> = {};
+  if ("decorators" in type) {
+    for (const decorator of type.decorators) {
+      // only process explicitly defined decorators
+      if (decorator.definition) {
+        const decoratorName = `${decorator.definition?.namespace ? getNamespaceFullName(decorator.definition.namespace) + "." : ""}${decorator.definition?.name}`;
+        // white list filtering
+        if (!context.decoratorsWhiteList) {
+          continue;
+        }
+        let found = false;
+        for (const item of context.decoratorsWhiteList) {
```

Comment:
i think it would be better to use `context.decoratorsWhiteList.filter(x =>x === decoratorName || new RegExp(x).test(decoratorName))`, or something like that, i haven't tested it out